JayJayson84/microservice.harness
# Feature/base

# Description

- [Add example contracts](https://github.com/JayJayson84/microservice.harness/commit/3cffd16298965c0650319a21efbc4a70c963121a)
- [Add example consumer](https://github.com/JayJayson84/microservice.harness/commit/e7f738f8546d19cfaa337bf15b10ea0c5a7658d1)
- [Add example producer](https://github.com/JayJayson84/microservice.harness/commit/e344d0b1d7d95b0c3f1c8f23fac594585971db44)
- [Add RabbitMQ configuration](https://github.com/JayJayson84/microservice.harness/commit/af3f2836b482faab152fa6393725541af8ad5b5f)
- [Add docker compose and env variables](https://github.com/JayJayson84/microservice.harness/commit/c557617f13e77c840dc57068ab1164449e061292)
- [Add debug configurations for VSCode](https://github.com/JayJayson84/microservice.harness/commit/2258891a534f19e093d47b3ee705622a02ac7906)
- [Add Solution](https://github.com/JayJayson84/microservice.harness/commit/8bbaa23e226e1144934cf3027c5cb18cdfa280e7)

## Where should the reviewer start?

1. Clone the branch.
1. Build and run the project in VSCode using the `Docker-Example-Producer` debug profile.

## How should this be tested?

[Swagger]
1. Open the Swagger URL for the Producer `http://localhost:5000/swagger`:
    1. Confirm Swagger loads the endpoints below.
    2. Confirm the endpoints result in a 200 OK response and JSON result for both endpoints.

[Postman]
1. Make a GET request in Postman to endpoint: `http://localhost:5000/api/Simple/CheckEndpoint`
    1. Verify a 200 OK response and JSON result.
1. Make a POST request in Postman to endpoint: `http://localhost:5000/api/Simple/GetResponse?value=test`
    1. Verify a 200 OK response and JSON result.

[Unit Tests]
Run the Unit Tests in the example Consumer and Producer projects and verify that they all pass.

*Tick all that apply:*

- [ ] Automation
- [ ] Regression check
